docs/tutorial/part04.rst

```diff
@@ -2,5 +2,437 @@
+    │   │   └── other.py
+    │   ├── tests.py
+    │   ├── urls.py
+    │   └── views.py
+    └── manage.py
+
+Now, how you actually query a graph database inside your Django project or apps?
+
+Using your models
+===================================
+
+Using your models is pretty standard. You usually just import the ones you need and use them, for
+example, in your Django views. Before we get to that, we need to learn the Neomodel Query API. This
+API will allow us to express queries to the database without having to write them in plain Cypher.
+
+We will learn to query using the Paradise Paper models we did before.
+To do that we will first use an instance of the python interpreter.
+
+The project python interpreter:
+---------------------------------------
+
+Let's open the python interpreter through our Django project ``manage.py`` which will import our
+project settings(remember you set the DATABASE_URL in there, this is needed to connect to the db).
+Also, it will try to use ipython or bpython if available.
+
+First let’s start opening the console, if necessary.
+
+Then make sure you are at the ``paradise_papers_search`` root directory
+where you created the Django project(where the ``manage.py`` module is).
+
+Run the command::
+
+        python manage.py shell
+
+
+With the python interpreter at hand, we can import our models and start to use them as soon as we
+execute the following python import commands::
+
+    from fetch_api.models import Entity
+    from fetch_api.models import Intermediary
+    from fetch_api.models import Officer
+    from fetch_api.models import Address
+    from fetch_api.models import Other
+
+Each of the models we just imported maps to a specific structure of a node label, property keys and
+relationship types in the PPGDB. Now we are actually ready to start exploring the
+Paradise Paper Graph Database through the Neomodel Query API.
+
+..  todo::
+    Put screenshot image of the console here.
+
+Neomodel Query API
+==================
+Each of your models has some properties and methods(inherited from StructureNode or DjangoNode)
+that help us to express queries to the Neo4j database.
+
+NodeSet and Nodes Neomodel objects
+--------------------------------------------------------
+A *NodeSet* object represents a set of nodes matching common query parameters or filters.
+
+A *Node* object is an instance of one of our models. That means we can access all the properties
+and methods defined on the model class. Each instance represents a single node in the database.
+
+The ``<Model>.nodes`` class property of each model store a NodeSet object. Each time we access
+this ``.nodes`` property we get a brand new nodeset object, which means we get nodeset without any
+filters applied. Initially, before applying any filters, this noseset represents all the nodes
+mapped under a model(nodes labeled with the same class name). For instance, ``Entity.nodes``
+contains all the nodes with the label Entity on the database.

```

Comment:
Sorry, a question here; is `contains` or `contain`?

Reply:
'Entity.nodes' is third person. So 'it contains'